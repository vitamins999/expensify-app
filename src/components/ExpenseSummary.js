import React from 'react'
import { connect } from 'react-redux'
import numeral from 'numeral'
import selectExpenses from '../selectors/expenses'
import getExpensesTotal from '../selectors/expenses-total'

export const ExpenseSummary = ({ numberOfExpenses, totalExpenses }) => {
    const expenseWord = numberOfExpenses === 1 ? 'expense' : 'expenses'
    const formattedTotal = numeral(totalExpenses / 100).format('$0,0.00')
    return (
        <div>
            {numberOfExpenses !== 0 ?
                <h1>Viewing {numberOfExpenses} {expenseWord}, totalling {formattedTotal}.</h1> : ''
            }
        </div>
    )
}

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters)

    return {
        numberOfExpenses: visibleExpenses.length,
        totalExpenses: getExpensesTotal(visibleExpenses)
    }
}

export default connect(mapStateToProps)(ExpenseSummary)