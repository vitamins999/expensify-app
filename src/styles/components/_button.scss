.button {
    background: #1c88bf;
    color: white;

    display: inline-block;

    border: none;
    font-weight: 300;
    font-size: $font-size-large;
    border-radius: 3px;
    padding: $s-size;
    text-decoration: none;
    line-height: 1;
}

.button--link {
    background: none;
}

.button--secondary {
    background: #888888;
}