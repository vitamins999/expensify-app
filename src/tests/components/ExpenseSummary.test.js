import React from 'react'
import { shallow } from 'enzyme'
import { ExpenseSummary } from '../../components/ExpenseSummary'

test('should correctly render expenses summary with 1 expense', () => {
    const wrapper = shallow(<ExpenseSummary numberOfExpenses={1} totalExpenses={123} />)
    expect(wrapper).toMatchSnapshot()
})

test('should correctly render expenses summary with multiple expenses', () => {
    const wrapper = shallow(<ExpenseSummary numberOfExpenses={23} totalExpenses={987654321} />)
    expect(wrapper).toMatchSnapshot()
})