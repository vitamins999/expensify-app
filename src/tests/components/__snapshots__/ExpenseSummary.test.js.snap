// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render expenses summary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
    , totalling 
    $1.23
    .
  </h1>
</div>
`;

exports[`should correctly render expenses summary with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    23
     
    expenses
    , totalling 
    $9,876,543.21
    .
  </h1>
</div>
`;
